system-tests/tests/src/test/java/org/eclipse/dataspaceconnector/identityhub/systemtests/VerifiableCredentialsIntegrationTest.java

```diff
@@ -0,0 +1,104 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.dataspaceconnector.identityhub.credentials.VerifiableCredentialsJwtService.VERIFIABLE_CREDENTIALS_KEY;
+
+@IntegrationTest
+@ExtendWith(EdcExtension.class)
+class VerifiableCredentialsIntegrationTest {
+
+    static final Faker FAKER = new Faker();
+    static final String HUB_URL = "http://localhost:8182/api/identity-hub";
+    static final String AUTHORITY_DID = "did:web:localhost:authority";
+    static final String PARTICIPANT_DID = "did:web:localhost:participant";
+    static final String AUTHORITY_PRIVATE_KEY_PATH = "resources/jwt/authority/private-key.pem";
+    static final ObjectMapper MAPPER = new ObjectMapper();
+    static final VerifiableCredential VC1 = VerifiableCredential.Builder.newInstance()
+            .id(FAKER.internet().uuid())
+            .credentialSubject(Map.of(
+                    FAKER.internet().uuid(), FAKER.lorem().word(),
+                    FAKER.internet().uuid(), FAKER.lorem().word()))
+            .build();
+
+    CommandLine cmd = IdentityHubCli.getCommandLine();
+    StringWriter out = new StringWriter();
+    StringWriter err = new StringWriter();
+
+    @BeforeEach
+    void setUp(EdcExtension extension) {
+        cmd.setOut(new PrintWriter(out));
+        cmd.setErr(new PrintWriter(err));
+
+        extension.setConfiguration(Map.of(
+                "edc.identity.hub.url", HUB_URL,
+                "edc.iam.did.web.use.https", "false"));
+    }
+
+    @Test
+    void push_and_get_verifiable_credentials(CredentialsVerifier verifier, DidResolverRegistry resolverRegistry) throws Exception {
+        assertIdentityHubIsEmpty();
+        addVerifiableCredential();
+        assertGetVerifiableCredential(verifier, resolverRegistry);
+    }
+
+    private void assertIdentityHubIsEmpty() throws JsonProcessingException {
+        int result = cmd.execute("-s", HUB_URL, "vc", "list");
+        assertThat(result).isEqualTo(0);
+
+        var claims = MAPPER.readValue(out.toString(), new TypeReference<List<Map<String, Object>>>() {});
+        assertThat(claims).describedAs("Identity Hub already contains Verifiable Credentials").isEmpty();
+    }
+
+    private void addVerifiableCredential() throws JsonProcessingException {
+        var json = MAPPER.writeValueAsString(VC1);
+        cmd.execute("-s", HUB_URL, "vc", "add", "-c", json, "-i", AUTHORITY_DID, "-b", PARTICIPANT_DID, "-k", AUTHORITY_PRIVATE_KEY_PATH);
+    }
+
+    private void assertGetVerifiableCredential(CredentialsVerifier verifier, DidResolverRegistry resolverRegistry) {
+        var didResult = resolverRegistry.resolve(PARTICIPANT_DID);
+        assertThat(didResult.succeeded()).isTrue();
+
+        var verifiedCredentials = verifier.getVerifiedCredentials(didResult.getContent());
+        assertThat(verifiedCredentials.succeeded()).isTrue();
+        var vcs = verifiedCredentials.getContent();
+        assertThat(vcs).containsKey(VC1.getId());
```

Comment:
from the context I understand we could be stricter here, and assert ONLY that key is present?

Reply:
I actually decided to relax the test and not verify that this is the single VC in the Identity Hub. This allows to run the test more than once without failures (which can be useful locally)